File: sdks/typescript/examples/golden_e2e.ts
#!/usr/bin/env ts-node
/**
 * Golden E2E (TypeScript) — ingest (optional), compile, session create/compile/history/replay.
 *
 * Prereqs:
 *   - AvocadoDB server running at http://127.0.0.1:8765
 *   - Current working directory is the project root you want to query (project = process.cwd())
 *
 * Run:
 *   cd sdks/typescript
 *   npm install && npm run build
 *   ts-node examples/golden_e2e.ts
 */
import { AvocadoDB, SessionManager } from '../src';

async function main() {
  const baseUrl = 'http://127.0.0.1:8765';
  const project = process.cwd();
  const db = new AvocadoDB(baseUrl);

  // Health check
  const reachable = await db.ping();
  if (!reachable) {
    console.error('Server not reachable at', baseUrl);
    process.exit(1);
  }

  // Stats
  const stats = await db.stats();
  console.log('Stats:', stats);
  if (stats.spans === 0) {
    console.log('No spans found. Ingest your project before running this example.');
    process.exit(0);
  }

  // Compile a query
  const query = 'What is Orion and how does it help APX?';
  const ws = await db.compile(query, { budget: 4000 });
  console.log('\nCompile summary:\n' + ws.summary());

  // Session flow
  const sm = new SessionManager(baseUrl, project);
  const session = await sm.createSession({ userId: 'golden-ts', title: 'Golden E2E (TS)' });
  console.log('\nSession created:', session.toString());

  const turn1 = await session.compile(query, { budget: 4000 });
  console.log('\nSession compile spans:', turn1.workingSet.spans.length);

  await session.addMessage('assistant', 'Context compiled. Proceeding.');
  const history = await session.getHistory(1500);
  console.log('\nHistory (truncated):\n', history.substring(0, 400), '...');

  const replay = await session.replay();
  console.log('\nReplay turns:', replay.turns.length);

  console.log('\nDone.');
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: sdks/python/examples/golden_e2e.py
from pathlib import Path
from avocado import AvocadoDB

def main() -> None:
    # Use daemon (HTTP mode). Ensure server is running separately.
    db = AvocadoDB()  # defaults to http://localhost:8765 and project=PWD

    # 1) Ingest a small document (idempotent; duplicates return 0 spans)
    sample = Path("golden.md")
    sample.write_text("# AvocadoDB\nDeterministic RAG with citations.\nMMR and hybrid search.\n")
    ingest = db.ingest(str(sample), sample.read_text())
    print("Ingest:", ingest)

    # 2) Ask (LLM disabled by default; returns deterministic context text)
    context = db.ask("What is AvocadoDB?", llm="none", budget=4000)
    print("Context preview:", context.splitlines()[:6])

    # 3) Sessions: create → compile → add message → history → replay
    s = db.create_session(user_id="golden", title="Golden E2E")
    result = s.compile("Explain determinism", budget=4000)
    print("Tokens used:", result["working_set"]["tokens_used"])
    s.add_message("assistant", "Determinism ensures same query → same context.")
    print("History:", s.get_history())
    replay = s.replay()
    print("Replay turns:", len(replay["turns"]))

if __name__ == "__main__":
    main()

